### PR TITLE
first draft of 21005-beta

### PR DESCRIPTION
**Pull request by jakub-pomykala**

### CODE REVIEW

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+---
+= Building Multipart payloads for JAX-RS client and services in Open Liberty 21.0.0.5-beta
+Jakub Pomykala <https://github.com/jakub-pomykala>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+With Open Liberty 21.0.0.5-beta you can now build multipart payloads for you JAX-RS client and services and also we now offer Jakarta EE 9 Support for `oauth-2.0` and `samlWeb-2.0`.
```
Comment (mbroz2) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: With Open Liberty 21.0.0.5-beta your can now build multipart payloads for **your** JAX-RS client and services.  We also now offer Jakarta EE 9 **support** for `oauth-2.0` and `samlWeb-2.0`.

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+The All Beta Features package includes 
+
+* <<jee9support, Jakarta EE 9 Support for oauth-2.0 and samlWeb-2.0>>
+* <<webservices, Build multipart payloads for your JAX-RS client and services>>.
+
+
+[#webservices]
+=== Build multipart payloads for your JAX-RS client and services
+
+Often, a RESTful service or client will need to send multiple disparate pieces of data in the same request, for example uploading a resume/CV with a picture, and text for name and address. This is usually done using `multipart/form-data`.
```
Comment (mbroz2) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: request, for example, uploading a resume/CV with a picture and text for name and address.

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+----
+
+Or take a look at our link:{url-prefix}/downloads/#runtime_betas[Downloads page].
+
+[#jakarta]
+== Jakarta EE 9 Beta Features package
+
+As of the 21.0.0.2-beta release, Open Liberty is the first vendor product to be Jakarta EE Web Profile 9.0 compatible. With the recent 21.0.0.3-beta release, Open Liberty is the first vendor product to be added to the link:https://jakarta.ee/compatibility/#tab-9[Jakarta EE Platform 9.0 compatibility list].
+
+or more information about Open Liberty's Jakarta EE 9 compatability, view our link:{url-prefix}/blog/2021/03/05/jakarta-ee-9-compatibility.html[Open Liberty beta is Jakarta EE 9 compatible] blog release.
```
Comment (mbroz2) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: This Open Liberty beta release comes complete with all of the previously released Jakarta EE9 API candidates. 
 **For** more information...

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes 
+
+* <<jee9support, Jakarta EE 9 Support for oauth-2.0 and samlWeb-2.0>>
```
Comment (mbroz2) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Swap order so jee9support is second

**Review by mbroz2 — DISMISSED** [review]
Why is "Support" capitalized in "Jakarta EE 9 Support"?  Seems to me like it should be lowercase in all instances

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 1 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 2 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 3 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 4 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 5]
```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+title: "Building Multipart payloads for JAX-RS client and services in Open Liberty 21.0.0.5-beta"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jakub-pomykala
+author_github: https://github.com/jakub-pomykala
+seo-title: Building Multipart payloads for JAX-RS client and services in Open Liberty 21.0.0.5-beta - OpenLiberty.io
+seo-description: With Open Liberty 21.0.0.5-beta you can now build multipart payloads for you JAX-RS client and services and also we now offer Jakarta EE 9 support for `oauth-2.0` and `samlWeb-2.0`.  
+blog_description: With Open Liberty 21.0.0.5-beta you can now build multipart payloads for you JAX-RS client and services and also we now offer Jakarta EE 9 support for `oauth-2.0` and `samlWeb-2.0`.
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]:
```suggestion
seo-description: With Open Liberty 21.0.0.5-beta you can now build multipart payloads for your JAX-RS client and services and also we now offer Jakarta EE 9 support for `oauth-2.0` and `samlWeb-2.0`.  
blog_description: With Open Liberty 21.0.0.5-beta you can now build multipart payloads for your JAX-RS client and services and also we now offer Jakarta EE 9 support for `oauth-2.0` and `samlWeb-2.0`.  
```

`and also we now offer` should probably be changed to something else.

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+* <<jee9support, Jakarta EE 9 support for oauth-2.0 and samlWeb-2.0>>s
+
+[#webservices]
+=== Build multipart payloads for your JAX-RS client and services
+
+Often, a RESTful service or client will need to send multiple disparate pieces of data in the same request, for example, uploading a resume/CV with a picture and text for name and address. This is usually done using `multipart/form-data`.
+
+While Liberty currently has APIs to enable users to _receive_ multipart/form-data payloads, it does not have any APIs to enable _send_ multipart payloads - until now. With the new `AttachmentBuilder` API, users can now send multipart requests from their JAX-RS clients or send multipart payloads as responses from their JAX-RS resources.
+
+To send a multipart request, you will need to enable the `jaxrs-2.0` or `jaxrs-2.1` feature. Presumably, if you are already using JAX-RS, one of these features is already enabled. To send a multipart payload, you must send an instance of `List<IAttachment>`.  Each object in that list represents a single attachment part.  Here is an example of creating and sending a multipart request from a JAX-RS client:
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]:
```suggestion
To send a multipart request, you will need to enable the `jaxrs-2.0` or `jaxrs-2.1` feature. Presumably, if you are already using JAX-RS, one of these features will already be enabled. To send a multipart payload, you must send an instance of `List<IAttachment>`.  Each object in that list represents a single attachment part.  Here is an example of creating and sending a multipart request from a JAX-RS client:
```

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+                                 .build());
+
+Response response = client.target(BLOG_SITE_URI)
+                          .request()
+                          .post(Entity.entity(attachments, MediaType.MULTIPART_FORM_DATA));
+----
+
+For more information vist:
+* link:https://tools.ietf.org/html/rfc7578[multipart/form-data RFC 7578]
+* link:JAVADOC[JavaDoc]
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc:75]: Does this link correctly? (Genuinely no idea)

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty.beta', name: 'openliberty-runtime', version: '[20.0.0.5-beta,)'
+}
+----
+
+Or take a look at our link:{url-prefix}/downloads/#runtime_betas[Downloads page].
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: This needs to be updated to the new standard that Joseph Cass has laid out as currently I this doesn't work afaik.

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+To try out these Jakarta EE 9 features on Open Liberty in a lightweight package, just update your build tools to pull the Open Liberty Jakarta EE 9 Beta Features package instead of the main release. The beta works with Java SE 15, Java SE 11, or Java SE 8.
+
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty.beta</groupId>
+    <artifactId>openliberty-jakartaee9</artifactId>
+    <version>20.0.0.5-beta</version>
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]:
```suggestion
    <version>21.0.0.5-beta</version>
```

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    <type>zip</type>
+</dependency>
+----
+
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty.beta', name: 'openliberty-jakartaee9', version: '[20.0.0.5-beta,)'
```
Comment (austin0) [inline posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]:
```suggestion
    libertyRuntime group: 'io.openliberty.beta', name: 'openliberty-jakartaee9', version: '[21.0.0.5-beta,)'
```

**Review by austin0 — CHANGES_REQUESTED** [review]

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 7 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc:75]: Yup, works fine

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 5 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Changed to ```With Open Liberty 21.0.0.5-beta you can now build multipart payloads for your JAX-RS client and services and also present in this release is Jakarta EE 9 support for `oauth-2.0` and `samlWeb-2.0`.  ```

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 6 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 8 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 10 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Thank you for spotting it

File: posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc (hunk 9 above)
Comment (jakub-pomykala) [reply posts/2021-04-19-JAXRS-multipart-payloads-21005-beta.adoc]: Done

**Review by austin0 — APPROVED** [review]

**Pull request merged**